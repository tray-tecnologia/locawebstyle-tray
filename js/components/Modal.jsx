import React from 'react';

class Modal extends React.Component {

    componentDidMount() {
        locastyle.modal.init();
        locastyle.general.init();
    }

    render() {
        return (
            <div>
                <div className="ls-modal" id="modal">
                    <div className="ls-modal-box">
                        <div className="ls-modal-header">
                            <button data-dismiss="modal">&times; </button>
                            <h4 className="ls-modal-title">Desabilitar aplicativo</h4>
                        </div>
                        <div className="ls-modal-body" id="myModalBody">
                            <p>Deseja desabilitar o aplicativo?</p>
                        </div>
                        <div className="ls-modal-footer ls-txt-right">
                            <button className="ls-btn ls-float-left" data-dismiss="modal">Fechar</button>
                            <button type="submit" className="ls-btn ls-btn-danger ls-btn-primary-danger" id="btn-modal" data-toggle-class="ls-display-none" data-target="#btn-modal, #btn-modal-disabled">
                                Desabilitar
                            </button>
                            <div>
                                <button id="btn-modal-disabled" className="ls-btn-danger ls-disabled ls-display-none" data-toggle-class="ls-display-none" data-target="#btn-modal, #btn-modal-disabled">
                                    Aguarde, desabilitando aplicativo...
                                </button>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}

export default Modal;
