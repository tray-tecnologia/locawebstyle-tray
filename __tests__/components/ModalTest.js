import React from 'react';
import renderer from 'react-test-renderer';

import Modal from '../../js/components/Modal.jsx';

locastyle.modal.init = function() {};
locastyle.general.init = function() {};

describe('Modal', () => {
    it('show modal"', () => {
        let modalTree = renderer.create(
            <Modal />
        ).toJSON();

        expect(modalTree).toMatchSnapshot();
    });
});
