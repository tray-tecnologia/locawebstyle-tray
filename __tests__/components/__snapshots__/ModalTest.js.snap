exports[`Modal show modal" 1`] = `
<div>
  <div
    className="ls-modal"
    id="modal">
    <div
      className="ls-modal-box">
      <div
        className="ls-modal-header">
        <button
          data-dismiss="modal">
          Ã— 
        </button>
        <h4
          className="ls-modal-title">
          Desabilitar aplicativo
        </h4>
      </div>
      <div
        className="ls-modal-body"
        id="myModalBody">
        <p>
          Deseja desabilitar o aplicativo?
        </p>
      </div>
      <div
        className="ls-modal-footer ls-txt-right">
        <button
          className="ls-btn ls-float-left"
          data-dismiss="modal">
          Fechar
        </button>
        <button
          className="ls-btn-danger"
          data-target="#btn-modal, #btn-modal-disabled"
          data-toggle-class="ls-display-none"
          id="btn-modal"
          type="submit">
          Desabilitar
        </button>
        <div>
          <button
            className="ls-btn-danger ls-disabled ls-display-none"
            data-target="#btn-modal, #btn-modal-disabled"
            data-toggle-class="ls-display-none"
            id="btn-modal-disabled">
            Aguarde, desabilitando aplicativo...
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
